src/core/thread/topology.hpp

```diff
@@ -862,6 +872,56 @@ class Child : public Neighbor, public IndirectSender::ChildInfo, public DataPoll
 
 #endif // #if OPENTHREAD_CONFIG_CHILD_SUPERVISION_ENABLE
 
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+    /**
+     * This method returns MLR state of an Ip6 address.
+     *
+     * @param[in] aAddressIndex  The Ip6 address index.
+     *
+     * @returns MLR state of the Ip6 address.
+     *
+     */
+    MlrState GetAddressMlrState(uint16_t aAddressIndex);
```

Comment:
I have an idea which I think can help simplify the code.

How about instead of passing `aAddressIndex` to the `Get/SetMlrState` methods, we require the caller to pass a reference to the `aAddress` entry itself.
-  We require `aAddress` reference to be the same reference as the one user does get from iterating over `Child` addresses (i.e. same reference that we get when we loop over addresses using `IterateIp6Addresses()`).
- Then inside `Child` class, we simply check the address of `aAddress` against `mIp6Address[]` to find an index which we can use to get to two masks `mMlrToRegisterMask` and `mMlrRegisteredMask`

I feel this model may be more straight-forward than current model of using an `AddressIterator` to go through addresses, converting the iterator to an index, then passing the index to `Child` methods get/set more info/state about the address of the child.

Reply:
updated accordingly